mozilla/blurts-server
# Add initial petition banner (MNTOR-3113)

# References:

Jira: The story is [MNTOR-3113](https://mozilla-hub.atlassian.net/browse/MNTOR-3113). The sub-tasks addressed in this PR are:
- [MNTOR-3323](https://mozilla-hub.atlassian.net/browse/MNTOR-3323)
- [MNTOR-3324](https://mozilla-hub.atlassian.net/browse/MNTOR-3324)
- [MNTOR-3325](https://mozilla-hub.atlassian.net/browse/MNTOR-3325)
- [MNTOR-3326](https://mozilla-hub.atlassian.net/browse/MNTOR-3326)

Telemetry is still TBD and will be added as a follow-up with [MNTOR-3337](https://mozilla-hub.atlassian.net/browse/MNTOR-3337).



# Description

Introduces the US data privacy petition banner.

As mentioned in the references we’ll need to add telemetry in a follow-up as soon as we have more info on which events should be recorded. The experiment `data-privacy-petition-banner`, which is responsible for showing this banner, is currently only enabled for `local` as there is a mutual dependency on the CSAT survey work that goes along with this banner in [MNTOR-3224](https://mozilla-hub.atlassian.net/browse/MNTOR-3224).

# Screenshot

<img width="970" alt="image" src="https://github.com/mozilla/blurts-server/assets/13835474/15ca242f-fb90-43e9-a997-768c15043663">

# How to test

1. Rebuild experiments: `npm run build-nimbus`
2. Navigate to the dashboard
3. US users should see the petition banner locally: Free users on the “action needed” and Plus users on the “fixed” tab. 

# Checklist (Definition of Done)

- [x] Jira ticket has been updated to match changes made during the development process.

[MNTOR-3113]: https://mozilla-hub.atlassian.net/browse/MNTOR-3113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3323]: https://mozilla-hub.atlassian.net/browse/MNTOR-3323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3324]: https://mozilla-hub.atlassian.net/browse/MNTOR-3324?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3325]: https://mozilla-hub.atlassian.net/browse/MNTOR-3325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3326]: https://mozilla-hub.atlassian.net/browse/MNTOR-3326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3337]: https://mozilla-hub.atlassian.net/browse/MNTOR-3337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3224]: https://mozilla-hub.atlassian.net/browse/MNTOR-3224?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
